packages/console/src/pages/SignInExperience/index.tsx

```diff
@@ -40,18 +45,28 @@ const SignInExperience = () => {
     }
   }, [data, reset]);
 
-  const onSubmit = handleSubmit(async (formData) => {
-    if (!data || isSubmitting) {
-      return;
-    }
-
+  const saveData = async () => {
     const updatedData = await api
       .patch('/api/sign-in-exp', {
-        json: signInExperienceParser.toRemoteModel(formData),
+        json: signInExperienceParser.toRemoteModel(getValues()),
       })
       .json<SignInExperienceType>();
     void mutate(updatedData);
     toast.success(t('application_details.save_success'));
+  };
+
+  const onSubmit = handleSubmit(async (formData) => {
+    if (!data || isSubmitting) {
+      return;
+    }
+
+    const formated = signInExperienceParser.toRemoteModel(formData);
+
+    if (compareSignInMethods(data, formated)) {
+      await saveData();
+    } else {
+      setDataToCompare(formated);
+    }
```

Comment:
What's the difference between these two branches? I'm a bit confused

Reply:
I have changed into `return` and added a comment. PTAL